src/Bucket.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import {
+  InputGroup,
+  InputGroupAddon,
+  InputGroupText,
+  Col,
+  Row,
+  Label,
+  Input
+} from 'reactstrap';
+
+class Bucket extends React.Component {
+  state = {
+    percentage: this.props.percentage
```

Comment:
I don't think you need any state management here and this could also be a functional component because you can use this prop directly.